noun/allocate.c

```diff
@@ -1628,22 +1628,128 @@ u3a_print_memory(c3_c* cap_c, c3_w wor_w)
 
   if ( byt_w ) {
     if ( gib_w ) {
-      fprintf(fil_f, "%s: GB/%d.%03d.%03d.%03d\r\n", 
+      fprintf(stderr, "%s: GB/%d.%03d.%03d.%03d\r\n",
           cap_c, gib_w, mib_w, kib_w, bib_w);
     }
     else if ( mib_w ) {
-      fprintf(fil_f, "%s: MB/%d.%03d.%03d\r\n", cap_c, mib_w, kib_w, bib_w);
+      fprintf(stderr, "%s: MB/%d.%03d.%03d\r\n", cap_c, mib_w, kib_w, bib_w);
     }
     else if ( kib_w ) {
-      fprintf(fil_f, "%s: KB/%d.%03d\r\n", cap_c, kib_w, bib_w);
+      fprintf(stderr, "%s: KB/%d.%03d\r\n", cap_c, kib_w, bib_w);
     }
     else if ( bib_w ) {
-      fprintf(fil_f, "%s: B/%d\r\n", cap_c, bib_w);
+      fprintf(stderr, "%s: B/%d\r\n", cap_c, bib_w);
     }
   }
-  u3_term_io_loja(0);
 }
 
+/* u3a_maid(): maybe print memory.
+*/
+c3_w
+u3a_maid(c3_o pri_o, c3_c* cap_c, c3_w wor_w)
+{
+  if ( c3y == pri_o ) {
+    u3a_print_memory(cap_c, wor_w);
+  }
+  return wor_w;
+}
+
+/* u3a_mark_road(): mark ad-hoc persistent road structures.
+*/
+c3_w
+u3a_mark_road(c3_o pri_o)
+{
+  c3_w tot_w = 0;
+  tot_w += u3a_maid(pri_o, "  namespace", u3a_mark_noun(u3R->ski.gul));
+  tot_w += u3a_maid(pri_o, "  trace stack", u3a_mark_noun(u3R->bug.tax));
+  tot_w += u3a_maid(pri_o, "  trace buffer", u3a_mark_noun(u3R->bug.mer));
+  tot_w += u3a_maid(pri_o, "  profile batteries", u3a_mark_noun(u3R->pro.don));
+  tot_w += u3a_maid(pri_o, "  profile doss", u3a_mark_noun(u3R->pro.day));
+  tot_w += u3a_maid(pri_o, "  new profile trace", u3a_mark_noun(u3R->pro.trace));
+  tot_w += u3a_maid(pri_o, "  memoization cache", u3h_mark(u3R->cax.har_p));
+  return   u3a_maid(pri_o, "total road stuff", tot_w);
+}
+
+static c3_c*
+_ca_print_box(u3a_box* box_u)
+{
+  if ( u3a_minimum < box_u->siz_w ) {
+    // fprintf(stderr, "print box 1\r\n");
+    return u3m_pretty(u3a_to_pug(u3a_outa(u3a_boxto(box_u))));
+  }
+  else {
+    // this might not be a cell, but we can use the struct to inspect further
+    //
+    u3a_cell* cel_u = (u3a_cell*)box_u;
+
+    // this condition potentially corresponds to an indirect atom of only one word
+    //
+    if ( (1 == (c3_w)cel_u->hed) &&
+         (0x80000000 & (c3_w)cel_u->tel) )
+    {
+      // this might not be right, but we know it's safe
```

Comment:
not clear what is "this".  Perhaps:  "this might not be an indirect atom, but we know it's safe to print as one".